src/components/Reviews.jsx

```diff
@@ -2,28 +2,105 @@ import { useEffect } from "react";
 import { getReviews } from "../requests";
 import { Link } from "react-router-dom";
 import { useState } from "react";
+import moment from "moment";
 
 const Reviews = ({ reviews, setReviews }) => {
   const [loading, setLoading] = useState(true);
+  const [sort, setSort] = useState(null);
   useEffect(() => {
     getReviews().then((data) => {
-      setReviews(data);
-      setLoading(false);
+      if (sort === "newest-oldest" || sort === null) {
+        setReviews(
+          [...data].sort(
+            (a, b) =>
+              moment(b.created_at).format("YYYYMMDDHHmmss") -
+              moment(a.created_at).format("YYYYMMDDHHmmss")
+          )
+        );
+        setLoading(false);
+      }
+      if (sort === "oldest-newest") {
+        setReviews(
+          [...data].sort(
+            (a, b) =>
+              moment(a.created_at).format("YYYYMMDDHHmmss") -
+              moment(b.created_at).format("YYYYMMDDHHmmss")
+          )
+        );
+      }
+      if (sort === "votes_asc") {
+        setReviews([...data].sort((a, b) => a.votes - b.votes));
+      }
+      if (sort === "votes_desc") {
+        setReviews([...data].sort((a, b) => b.votes - a.votes));
+      }
+      if (sort === "comments_asc") {
+        setReviews([...data].sort((a, b) => a.comment_count - b.comment_count));
+      }
+      if (sort === "comments_desc") {
+        setReviews([...data].sort((a, b) => b.comment_count - a.comment_count));
+      }
```

Comment:
similarly here, we want our filtered/sorted results to be coming from the API request itself